apps/dc_tools/tests/test_sns_publishing.py

```diff
@@ -0,0 +1,261 @@
+STAC_DATA: str = "ga_ls5t_nbart_gm_cyear_3_x30y14_1999--P1Y_final.stac-item.json"
+
+
+@pytest.fixture
+def sqs_message():
+    with TEST_DATA_FOLDER.joinpath(STAC_DATA).open("r") as f:
+        body = json.load(f)
+    message = {
+        "Type": "Notification",
+        "MessageId": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxxxxxxxxx",
+        "TopicArn": "arn:aws:sns:ap-southeast-2:xxxxxxxxxxxxxxxxx:test-topic",
+        "Subject": "Amazon S3 Notification",
+        "Message": json.dumps(body),
+        "Timestamp": "2020-08-21T08:28:45.921Z",
+        "SignatureVersion": "1",
+        "Signature": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+        "SigningCertURL": "https://sns.ap-southeast-2.amazonaws.com/SimpleNotificationService-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.pem",
+        "UnsubscribeURL": "https://sns.ap-southeast-2.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:ap-southeast-2:xxxxxxxxxxxxxxx:test-topic:xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxxxx",
+    }
+    return message
+
+
+@mock_sns
+@mock_sqs
+def test_sqs_publishing(aws_credentials, aws_env, sqs_message):
+    "Test that actions are published with sqs_to_dc"
+    sns = boto3.client("sns")
+    sqs = boto3.client("sqs")
+
+    topic_name = "test-topic"
+    queue_name = "test-queue"
+    input_queue_name = "input-queue"
+
+    input_queue = sqs.create_queue(QueueName=input_queue_name)
+    sqs.send_message(
+        QueueUrl=input_queue.get("QueueUrl"),
+        MessageBody=json.dumps(sqs_message),
+    )
+
+    topic = sns.create_topic(Name=topic_name)
+    queue = sqs.create_queue(QueueName=queue_name)
+    attrs = sqs.get_queue_attributes(
+        QueueUrl=queue.get("QueueUrl"), AttributeNames=["All"]
+    )
+    queue_arn = attrs["Attributes"]["QueueArn"]
+    sns_arn = topic.get("TopicArn")
+    sns.subscribe(
+        TopicArn=sns_arn,
+        Protocol="sqs",
+        Endpoint=queue_arn,
+    )
+
+    runner = CliRunner()
+    result = runner.invoke(
+        sqs_cli,
+        [
+            input_queue_name,
+            "ga_ls5t_nbart_gm_cyear_3",
+            "--skip-lineage",
+            "--statsd-setting",
+            "localhost:8125",
+            "--no-sign-request",
+            "--update-if-exists",
+            "--stac",
+            f"--publish-action={sns_arn}",
+        ],
+    )
+
+    messages = sqs.receive_message(
+        QueueUrl=queue.get("QueueUrl"),
+    )["Messages"]
+    assert result.exit_code == 0
+    assert len(messages) == 1
+    message_attrs = json.loads(messages[0]["Body"]).get("MessageAttributes")
+    assert message_attrs["action"].get("Value") == "ADDED"
+    assert message_attrs["product"].get("Value") == "ga_ls5t_nbart_gm_cyear_3"
+    assert message_attrs["maturity"].get("Value") == "final"
+
+
+@mock_sns
+@mock_sqs
+def test_sqs_publishing_archive(aws_credentials, aws_env, sqs_message):
```

Comment:
The only test case this doesn't cover is the scenario of using `--archive-less-mature`  + `--publish-action`. Might be worth adding, because it's the main thing I'll immediately be using it for

Reply:
Or is this essentially covered via functionality of `---archive`?

Reply:
Added